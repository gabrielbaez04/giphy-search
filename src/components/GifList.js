import React from 'react';
import Gif from './Gif';
import PropTypes from 'prop-types';

const GifList = ({ gifs, onGifSelect }) => {
  return (
    <div className="gif-list">
      {
        gifs.map((image) => {
          return <Gif key={image.id} gif={image} onGifSelect={onGifSelect} />
        })
      }
    </div>
  );
};
GifList.propTypes = {
  gifs: PropTypes.array,
  onGifSelect: PropTypes.func.isRequired
}
export default GifList;