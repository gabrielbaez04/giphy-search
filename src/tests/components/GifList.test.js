import { shallow } from 'enzyme';
import GifList from '../../components/GifList';
import React from 'react';

let gifs;
beforeEach(() => {
    gifs = [
        { id: 1, title: "test", images: { downsized_still: { url: "test.url.com" } } },
        { id: 2, title: "test", images: { downsized_still: { url: "test.url.com" } } },
        { id: 3, title: "test", images: { downsized_still: { url: "test.url.com" } } },
    ]
});
test('should render GifList correctly', () => {
    const wrapper = shallow(<GifList gifs={gifs} onGifSelect={() => { }} />);
    expect(wrapper).toMatchSnapshot();
});

test('should render the amount of Gifs in the GifList', () => {
    const wrapper = shallow(<GifList gifs={gifs} onGifSelect={() => { }} />);
    expect(wrapper.find('Gif').length).toBe(gifs.length);
});

