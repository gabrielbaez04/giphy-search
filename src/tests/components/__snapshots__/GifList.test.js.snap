// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GifList correctly 1`] = `
<div
  className="gif-list"
>
  <Gif
    gif={
      Object {
        "id": 1,
        "images": Object {
          "downsized_still": Object {
            "url": "test.url.com",
          },
        },
        "title": "test",
      }
    }
    key="1"
    onGifSelect={[Function]}
  />
  <Gif
    gif={
      Object {
        "id": 2,
        "images": Object {
          "downsized_still": Object {
            "url": "test.url.com",
          },
        },
        "title": "test",
      }
    }
    key="2"
    onGifSelect={[Function]}
  />
  <Gif
    gif={
      Object {
        "id": 3,
        "images": Object {
          "downsized_still": Object {
            "url": "test.url.com",
          },
        },
        "title": "test",
      }
    }
    key="3"
    onGifSelect={[Function]}
  />
</div>
`;
